libraries/AP_Generator/AP_Generator_RichenPower.cpp

```diff
@@ -266,8 +327,12 @@ bool AP_Generator_RichenPower::pre_arm_check(char *failmsg, uint8_t failmsg_len)
     }
     if (last_reading.seconds_until_maintenance == 0) {
         snprintf(failmsg, failmsg_len, "requires maintenance");
+    }
+    if (SRV_Channels::get_channel_for(SRV_Channel::k_richenpower_control) == nullptr) {
```

Comment:
It's an existing issue perhaps but I just want to reiterate that I think the servo channel should be k_generator_control instead of the more specific k_richenpower_control.